pymc_experimental/model/marginal_model.py

```diff
@@ -244,6 +249,134 @@ def marginalize(self, rvs_to_marginalize: Union[TensorVariable, Sequence[TensorV
         # Raise errors and warnings immediately
         self.clone()._marginalize(user_warnings=True)
 
+    def unmarginalize(self, rvs_to_unmarginalize):
+        for rv in rvs_to_unmarginalize:
+            self.marginalized_rvs.remove(rv)
+            self.register_rv(rv, name=rv.name)
+
+    def recover_marginals(
+        self, idata, var_names=None, include_samples=False, extend_inferencedata=True
```

Comment:
I would perhaps default `include_samples=True` by default. Also maybe a different name? `return_samples` or just `sample`?

Reply:
The discrete samples are often not as good for understanding the tail probabilities

This can be seen by comparing the changepoint's logps vs the discrete samples discussed on https://mc-stan.org/docs/stan-users-guide/change-point.html

Reply:
I know that but I still think it doesn't hurt to include by default. logits are not something many users grok intuitively

Reply:
Updated name